app/script/controllers/my.profile.controller.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Created by sa on 2016.12.22..
+ */
+(function () {
+    angular.module('app')
+        .controller('myProfileController',function (apiService, $scope, $location) {
+                apiService.getUserId()
+                    .then(function (response) {
+                        $scope.myProfile=response.data;
+                        $scope.right=response.data.rights;
+                    });
+                $scope.myProfileUpdate=function () {
+
+                    apiService.myProfileUpdate($scope.myProfile)
+                        .then(function (response) {
+                            $scope.myProfile=response.data;
+                        })
+                };
+
+            $scope.checkPhone = function(data) {
+                var re = new RegExp(/^\(?(\d{2})\)?[- ]?(\d{3})[- ]?(\d{4})$/);
```

Comment:
I don't think people put parentheses around '06', we don't have to check it, and also, you don't use the groups you matched that's unnecessary too.